### PR TITLE
Finish all user stories for v2.0

### PR DESCRIPTION
**Pull request by hshiah**
Finish all user stories for v2.0

### CODE REVIEW

File: src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import seedu.financialplanner.cashflow.CashflowList;
+import seedu.financialplanner.goal.Goal;
+import seedu.financialplanner.enumerations.ExpenseType;
+public class MarkGoalCommand extends Command {
     private final int index;
-    public UnmarkReminderCommand(RawCommand rawCommand) throws IllegalArgumentException {
+    public MarkGoalCommand(RawCommand rawCommand) throws IllegalArgumentException {
         String stringIndex;
-        if (rawCommand.args.size() == 1) {
+        if(rawCommand.args.size() == 1) {
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java]:
```suggestion
        if (rawCommand.args.size() == 1) {
```

File: src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java [hunk 2]
```diff
@@ -19,7 +23,7 @@ public UnmarkReminderCommand(RawCommand rawCommand) throws IllegalArgumentExcept
         if (index == 0) {
             throw new IllegalArgumentException("Index must be within the list");
         }
-        if (index > ReminderList.getInstance().list.size()+1){
+        if (index > WishList.getInstance().list.size()+1){
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java:26]:
```suggestion
        if (index > WishList.getInstance().list.size() + 1){
```

File: src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java [hunk 3]
```diff
@@ -31,7 +35,9 @@ public UnmarkReminderCommand(RawCommand rawCommand) throws IllegalArgumentExcept
 
     @Override
     public void execute() {
-        ReminderList.getInstance().list.get(index-1).unmark();
-        Ui.getInstance().showMessage("You have unmarked "+ReminderList.getInstance().list.get(index-1));
+        Goal goal = WishList.getInstance().list.get(index-1);
+        goal.markAsDone();
+        Ui.getInstance().showMessage("You have achieved " + goal + System.lineSeparator() +"Congratulations!");
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java]:
```suggestion
        Ui.getInstance().showMessage("You have achieved " + goal + System.lineSeparator() + "Congratulations!");
```

File: src/main/java/seedu/financialplanner/enumerations/ExpenseType.java [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import java.util.Arrays;
 
 public enum ExpenseType {
-    DINING, ENTERTAINMENT, SHOPPING, TRAVEL, INSURANCE, NECESSITIES, OTHERS;
+    DINING, ENTERTAINMENT, SHOPPING, TRAVEL, INSURANCE, NECESSITIES, OTHERS, GOAL;
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/enumerations/ExpenseType.java]: can the goal be one of the expensetypes? ie. buy car would be under 'others'. so goal shouldnt be one of the types

**Review by NeoMinWei — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/enumerations/ExpenseType.java (hunk 4 above)
Comment (hshiah) [reply src/main/java/seedu/financialplanner/enumerations/ExpenseType.java]: Make sense.

File: src/main/java/seedu/financialplanner/commands/DeleteGoalCommand.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import seedu.financialplanner.utils.Ui;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import seedu.financialplanner.goal.Goal;
+public class DeleteGoalCommand extends Command {
+    private static final Logger logger = Logger.getLogger("Financial Planner Logger");
+    private final int index;
+    public DeleteGoalCommand(RawCommand rawCommand) throws IllegalArgumentException {
+        String stringIndex;
+        if (rawCommand.args.size() == 1){
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/commands/DeleteGoalCommand.java]:
```suggestion
        if (rawCommand.args.size() == 1) {
```

File: src/main/java/seedu/financialplanner/goal/Goal.java
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/goal/Goal.java:1]: Can add functions to save and load goal and reminder from storage

**Review by ryan1604 — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/goal/Goal.java
Comment (hshiah) [reply src/main/java/seedu/financialplanner/goal/Goal.java:1]: Sure, I will do it in v2.1 as enhancement.

Comment (ryan1604) [reply src/main/java/seedu/financialplanner/goal/Goal.java:1]: Thank you

**Pull request merged**